packages/syft/src/syft/service/job/job_service.py

```diff
@@ -57,6 +61,27 @@ def get_all(self, context: AuthedServiceContext) -> Union[List[Job], SyftError]:
             res = res.ok()
             return res
 
+    @service_method(
+        path="job.get_by_user_code_id",
+        name="get_by_user_code_id",
+        roles=DATA_SCIENTIST_ROLE_LEVEL,
+    )
+    def get_by_user_code_id(
+        self, context: AuthedServiceContext, user_code_id: UID
+    ) -> Union[List[Job], SyftError]:
+        all_jobs = self.stash.get_all(context.credentials)
+        if all_jobs.is_err():
+            return SyftError(message=all_jobs.err())
+
+        all_jobs = all_jobs.ok()
+
+        res = []
+        for job in all_jobs:
```

Comment:
this should probably be a stash method that does the querying, in that case you also search by unique keys I think